### PR TITLE
New DocumentWriter constructors

### PR DESCRIPTION
**Pull request by benmccallum**
Simplifies/unifies instantiation of a `DocumentWriter`, especially in the server project, as this aligns with the configurability of the deserializers.

### CODE REVIEW

File: src/GraphQL.SystemTextJson/DocumentWriter.cs [hunk 1]
```diff
@@ -16,20 +17,36 @@ public DocumentWriter()
         }
 
         public DocumentWriter(bool indent)
-            : this(new JsonSerializerOptions { WriteIndented = indent, PropertyNamingPolicy = JsonNamingPolicy.CamelCase })
+            : this(GetDefaultSerializerSettings(indent))
         {
         }
 
-        public DocumentWriter(JsonSerializerOptions options)
+        public DocumentWriter(Action<JsonSerializerOptions> configureSerializerSettings)
```
Comment (sungam3r) [inline src/GraphQL.SystemTextJson/DocumentWriter.cs]:
```suggestion
        public DocumentWriter(Action<JsonSerializerOptions> configureSerializerOptions)
```

**Comment by benmccallum** [comment]
Sorted, merge away.

File: src/GraphQL.SystemTextJson/DocumentWriter.cs [hunk 2]
```diff
@@ -16,20 +17,36 @@ public DocumentWriter()
...
+
+        private void ConfigureOptions()
+        {
             if (!_options.Converters.Any(c => c.CanConvert(typeof(ExecutionResult))))
             {
                 _options.Converters.Add(new ExecutionResultJsonConverter());
             }
         }
 
+        private static JsonSerializerOptions GetDefaultSerializerSettings(bool indent)
```
Comment (sungam3r) [inline src/GraphQL.SystemTextJson/DocumentWriter.cs]:
```suggestion
        private static JsonSerializerOptions GetDefaultSerializerOptions(bool indent)
```

Sorry, one more change.

Comment (benmccallum) [reply src/GraphQL.SystemTextJson/DocumentWriter.cs]: Done.

File: src/GraphQL.SystemTextJson/DocumentWriter.cs [hunk 3]
```diff
@@ -16,20 +17,36 @@ public DocumentWriter()
...
-            : this(new JsonSerializerOptions { WriteIndented = indent, PropertyNamingPolicy = JsonNamingPolicy.CamelCase })
+            : this(GetDefaultSerializerSettings(indent))
         {
         }
 
-        public DocumentWriter(JsonSerializerOptions options)
+        public DocumentWriter(Action<JsonSerializerOptions> configureSerializerOptions)
         {
-            _options = options;
+            _options = GetDefaultSerializerSettings(indent: false);
```
Comment (sungam3r) [inline src/GraphQL.SystemTextJson/DocumentWriter.cs]:
```suggestion
            _options = GetDefaultSerializerOptions(indent: false);
```

File: src/GraphQL.SystemTextJson/DocumentWriter.cs [hunk 4]
```diff
@@ -16,20 +17,36 @@ public DocumentWriter()
         }
 
         public DocumentWriter(bool indent)
-            : this(new JsonSerializerOptions { WriteIndented = indent, PropertyNamingPolicy = JsonNamingPolicy.CamelCase })
+            : this(GetDefaultSerializerSettings(indent))
```
Comment (sungam3r) [inline src/GraphQL.SystemTextJson/DocumentWriter.cs]:
```suggestion
            : this(GetDefaultSerializerOptions(indent))
```

**Comment by benmccallum** [comment]
I did a proper refactor this time :P

**Pull request merged**